pints/_abc/_abc_smc.py

```diff
@@ -0,0 +1,176 @@
+                    # check if theta_s_s is possible under the prior
+                    # sample again if not
+                self._xs.append(theta_s_s)
+        self._ready_for_tell = True
+        return self._xs
+
+    def tell(self, fx):
+        """ See :meth:`ABCSampler.tell()`. """
+        if not self._ready_for_tell:
+            raise RuntimeError('tell called before ask.')
+        self._ready_for_tell = False
+        if isinstance(fx, list):
+            accepted = [a < self._threshold for a in fx]
+            if sum(accepted) > 0:
+                if self._t == len(self._e_schedule) - 1:
+                    return [self._xs[c].tolist() for c, x in
+                            enumerate(accepted) if x]
+
+                self._accepted_count += sum(accepted)
+                self._samples[self._t].extend(
+                    [self._xs[c].tolist() for c, x in enumerate(accepted) if x]
+                )
+
+                if self._accepted_count >= self._nr_samples:
+                    self._advance_time()
+            return None
+        else:
+            if fx < self._threshold:
+                self._accepted_count += 1
+                if self._t == len(self._e_schedule) - 1:
+                    return self._xs
+                self._samples[self._t].append(self._xs)
+                if self._accepted_count >= self._nr_samples:
+                    self._advance_time()
+            return None
+
+    def _advance_time(self):
+        t = self._t
+        if t == 0:
+            self._weights.append(
+                np.full(self._accepted_count, 1 / self._accepted_count))
+        else:
+            unnorm_weights = self._calculate_weights(
+                self._samples[t], self._samples[t - 1], self._weights[t - 1])
+            # Normalise weights
+            normal = sum(unnorm_weights)
+            self._weights.append([w / normal for w in unnorm_weights])
+
+        self._samples.append([])
+        self._accepted_count = 0
+        self._t += 1
+        self._threshold = self._e_schedule[self._t]
+        print(
+            "Trying t=" + str(self._t)
+            + ", threshold=" + str(self._threshold))
+
+    def _calculate_weights(self, new_samples, old_samples, old_weights):
+        new_weights = []
+        for i in range(0, self._accepted_count):
+            prior_prob = np.exp(self._log_prior(new_samples[i]))
+
+            mw = [old_weights[j] * np.exp(self._perturbation_kernel(
+                np.subtract(new_samples[i], old_samples[j])))
+                for j in range(len(old_samples))]
+
+            w = prior_prob / sum(mw)
+            new_weights.append(w)
+        return new_weights
+
+    def set_threshold_schedule(self, schedule):
+        """
+        Sets a schedule for the threshold error distance that determines if a
+        sample is accepted (if error < threshold).
+        Schedule should be a list of epsilon values
```

Comment:
1. Needs to be a new paragraph (see comment above).
2. What's an "epsilon value" ?